nvdaHelper/vbufBase/backend.cpp

```diff
@@ -147,50 +147,36 @@ bool VBufBackend_t::invalidateSubtree(VBufStorage_controlFieldNode_t* node) {
 	LOG_DEBUG(L"Invalidating node "<<node->getDebugInfo());
 	this->lock.acquire();
 	bool needsInsert=true;
-	for(VBufStorage_controlFieldNodeList_t::iterator i=invalidSubtreeList.begin();i!=invalidSubtreeList.end();) {
+	auto i=pendingInvalidSubtreesList.begin();
+	for(;i!=pendingInvalidSubtreesList.end();++i) {
 		VBufStorage_fieldNode_t* existingNode=*i;
 		if(node==existingNode) { //Node already invalidated
 			LOG_DEBUG(L"Node already invalidated");
 			needsInsert=false;
 			break;
-		} else if(isDescendantNode(existingNode,node)) { //An ancestor is already invalidated
-			LOG_DEBUG(L"An ancestor is already invalidated, returning");
-			needsInsert=false;
+		} else if(isDescendantNode(node,existingNode)) { // a descendant is already inserted
+			LOG_DEBUG(L"Inserting invalid subtree before first descendant");
 			break;
-		} else if(isDescendantNode(node,existingNode)) { //A descedndant has been invalidated
-			LOG_DEBUG(L"removing a descendant node from the invalid nodes");
-			//Remove the old descendant

```

Comment:
We no longer remove descendants in invalidateSubtree. That's fine for gecko_ia2 because while rendering a parent, reuseExistingNode will un-invalidate a child node if it isn't reused. If it's a grandchild or deeper, we do want update to re-render it anyway. However, this is problematic for other backends which don't use reuseExistingNode because they always re-render all descendants in fillVBuf, but those descendants will then get re-rendered again by update. Either:

1. reuseExistingNode must be used by all backends. This means fixing the bugs with mshtml. As discussed, this is probably due to updateAncestor.
2. This new behaviour of invalidateSubtree where it keeps descendants must only be used for gecko_ia2.

Option 1 is preferable, but failing that, option 2 will be necessary.